frontend/src/components/DropdownBox/DropdownBox.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { useState } from 'react';
+import './DropdownBox.css';
+
+type DropdownBoxProps = {
+  placeHolderText: string;
+  dataSource: Array<string>;
+  isOpen: boolean;
+  confirmText: string;
+};
+
+const DropdownBox = ({
+  placeHolderText,
+  dataSource,
+  isOpen,
+  confirmText,
+}: DropdownBoxProps) => {
+  const [isDropdownBoxOpen, setIsDropdownBoxOpen] = useState(isOpen);
+  const [selectedItems, setSelectedItems] = useState(new Set<string>());
+
+  const toggleDropdownBox = () => {
+    setIsDropdownBoxOpen(!isDropdownBoxOpen);
+  };
+
+  const toggleSelectItem = (itemName: string) => {
+    const newSelectedItems = new Set(selectedItems);
+    if (newSelectedItems.has(itemName)) {
+      newSelectedItems.delete(itemName);
+    } else {
+      newSelectedItems.add(itemName);
+    }
+    setSelectedItems(newSelectedItems);
+  };
+
+  return (
+    <div className="dropdown-wrapper">
+      <button
+        type="button"
+        className="dropdown-header"
+        onClick={toggleDropdownBox}
+      >
+        <div className="dropdown-header-contents">
+          {placeHolderText}
+          {'      '}
+          {isDropdownBoxOpen ? '⇈' : '⇊'}
+        </div>
+      </button>
+
+      {isDropdownBoxOpen && (
+        <>
+          <div className="dropdown-list">
```

Comment:
I noticed the dropdown box was custom styled and this is okay if there is a significant change from the native HTML styling but since they are very similar (essentially the same) lets stick to using native HTML elements as they are designed to be accessible and cross platform 

A suggestion is to map out everything and use `input type = checkbox`

Example 
```
      {isDropdownBoxOpen && (
        <div className="dropdown-list">
          {dataSource.map((itemName) => (
            <label key={itemName} className="dropdown-list-item">
              <input
                type="checkbox"
                checked={selectedItems.has(itemName)}
                onChange={() => toggleSelectItem(itemName)}
              />
              <span className="item-name">{itemName}</span>
            </label>
          ))}
```